docs/transaction-verification.md

```diff
@@ -0,0 +1,169 @@
+On the other hand, if the transaction passed the `initial verification`, but at the time it was received it had an unknown identity provider or set of anonymity revokers then the credential deployment will be verified again.
+If this results in a successful verification then the account is created on the chain otherwise the transaction is deemed invalid. Nb. Only credential deployments received as part of a block where the identity provider or the anonymity revokers were not deemed valid will be re-verified by the `Scheduler`. Credential deployments received individually have always been deemed valid in the `initial verification` in order to ever reach the point of execution.
+ 
+#### Chain Update
+Before executing a chain update the following is always checked:
+ 
+1. `expiry` The transaction must never be expired and if it is (expired) then it will be deemed invalid.
+2. `sequential sequence number` It is verified that the sequence number of the transaction is the next one.
+If the `sequence number` is not the next one then the transaction is invalid and will not be executed.
+ 
+The `Scheduler` will now use the cached verification result computed in the `initial verification`. 
+If the transaction passed the `initial verification` but the signatures could not be validated then it is checked now and based on the result it will be deemed valid or invalid.
+ 
+On the other hand, if the transaction did pass the `initial verification` the cached verification result contains a hash of the public keys used for verifying the signature. It is asserted that this hash corresponds to the current set of authorized keys. If the authorized keys are the same then the transaction is deemed valid. Otherwise we re-verify the signature on the transaction as the authorized keys might have changed since the transaction was first processed, if the 
+signature can be verified the transaction is accepted otherwise it is rejected.
+ 
+#### Transaction
+Before executing a `normal transaction` the following is always checked:
+ 
+1. `remaining block energy` There must be enough energy left in the current block, if this is not the case then the transaction will be considered invalid.
+2. `expiry` The transaction may not be expired.
+3. `sender exists` The sender of the transaction must exist on the chain. If the sender does not exist then the transaction is deemed invalid.
+4. `senders balance` The sender must have enough funds to cover the transfer. If this is not the case then the transaction is deemed invalid.
+5. `sequential nonce` The _next_ account nonce of the sender account must match the nonce in the transaction. If not, then
+the transaction invalid.
+ 
+If the transaction was received as part of a block and did only partially pass the `initial verification` then the transaction is re-verified. This verification must pass _successfully_ otherwise the transaction is deemed invalid.
```

Comment:
I don't understand what "partially pass the initial verification" means. Can you elaborate?

Reply:
Rephrased the sentence.